packages/components/src/Modal/Modal.rebuilt.test.tsx

```diff
@@ -76,41 +76,153 @@ describe("Composable Modal", () => {
       "secondary destructive",
     );
   });
-  it('modal contains aria role of "dialog"', async () => {
+
+  it("should render the modal header with custom content", () => {
     render(
-      <Modal.Provider open={true}>
+      <Modal.Provider open={true} modalLabelledBy="custom-header">
         <Modal.Content>
-          <Modal.Header title="Modal Title" />
+          <Modal.Header>
+            <span id="custom-header">Custom Header Content</span>
+          </Modal.Header>
         </Modal.Content>
       </Modal.Provider>,
     );
-    expect(await screen.findByRole("dialog")).toBeInTheDocument();
+    expect(screen.getByRole("dialog")).toHaveAttribute(
+      "aria-labelledby",
+      "custom-header",
+    );
+    expect(screen.getByRole("dialog")).toHaveTextContent(
+      "Custom Header Content",
+    );
   });
 
-  it("names the dialog from title via aria-labelledby when ariaLabel is not provided", async () => {
-    render(
-      <Modal.Provider open={true}>
-        <Modal.Content>
-          <Modal.Header title="Billing Settings" />
-        </Modal.Content>
-      </Modal.Provider>,
-    );
+  describe("accessibility", () => {
+    it('modal contains aria role of "dialog"', async () => {
+      render(
+        <Modal.Provider open={true}>
+          <Modal.Content>
+            <Modal.Header title="Modal Title" />
+          </Modal.Content>
+        </Modal.Provider>,
+      );
+      expect(await screen.findByRole("dialog")).toBeInTheDocument();
```

Comment:
we actually did this in the test above, but since it's just always open  we don't need await/findBy 

it's not really doing anything to use findBy but let's just handle it consistently. my preference would be no findBy if we have nothing to wait for.

Reply:
since we're dealing with animations, `toBeVisible` is preferable